### DISCUSSION TITLE
[Miscellaneous] devnet always downloads go dependencies

### DISCUSSION THREAD

**Discussion by karlb** [ethereum-optimism/developers#554] [Miscellaneous] [answered]
### Did you check the documentation?

- [X] Yes

### Did you check for duplicate questions?

- [X] Yes

### Issue Description

When doing changes to the op-stack implementation, the devnet is very useful. However, even after small changes, a `make devnet-up` takes rather long because go deps are downloaded again every time (those poor servers!). Is there a reason why the go module downloads are not cached or is it just that nobody got around to implementing the caching? Do you all live with the slow build times or am I missing a way to quickly iterate in the devnet?

**Comment by joohhnnn** [comment]
Would you mind posting the install logs here?  Thanks!

**Reply by karlb** [reply]
After doing a 1 line change in op-batcher, I get the following (attached) build log when doing `make devnet-up` on a system where I ran the devnet just before the change.
[devnet-up.log](https://github.com/user-attachments/files/16600323/devnet-up.log)

macOS Sonoma 14.6 (23G80)
Docker version 24.0.6, build ed223bc
using a forked Optimism version without changes to the docker setup: https://github.com/celo-org/optimism/commit/cb53122a2bf2a04f7d17bdeefe881b0faaada8e5
I can try again on a vanilla optimism if there have been any recent changes that might fix this.

**Reply by joohhnnn** [reply]
It might be caused by the settings of GITCOMMIT and GITDATE, but I'm not entirely sure. CC @opfocus

**Comment by anacrolix** [comment, marked as answer]
Hello, this should be fixed in https://github.com/ethereum-optimism/optimism/pull/11512. Let me know how it goes for you.

**Reply by karlb** [reply]
Looks good so far. Thanks!
